src/main/java/org/opentripplanner/openstreetmap/wayproperty/specifier/Condition.java

```diff
@@ -2,44 +2,31 @@
 
 import static org.opentripplanner.openstreetmap.wayproperty.specifier.Condition.MatchResult.EXACT;
 import static org.opentripplanner.openstreetmap.wayproperty.specifier.Condition.MatchResult.NONE;
+import static org.opentripplanner.openstreetmap.wayproperty.specifier.Condition.MatchResult.WILDCARD;
 
-import javax.annotation.Nonnull;
+import java.util.Arrays;
 import org.opentripplanner.openstreetmap.model.OSMWithTags;
 
 public sealed interface Condition {
-  @Nonnull
-  static MatchResult getMatchResult(OSMWithTags way, String opKey, String opValue) {
-    if (opValue.equals("*") && way.hasTag(opKey)) {
-      return MatchResult.WILDCARD;
-    } else if (way.matchesKeyValue(opKey, opValue)) {
-      return EXACT;
-    } else if (opValue.contains(":")) {
-      // treat cases like cobblestone:flattened as cobblestone if a more-specific match
-      // does not apply
-      var splitValue = opValue.split(":", 2)[0];
-      if (way.matchesKeyValue(opKey, splitValue)) {
-        return MatchResult.PREFIX;
-      } else {
-        return NONE;
-      }
-    } else {
-      return NONE;
-    }
-  }
+  boolean getMatches(OSMWithTags way, String opKey);
```

Comment:
This method is called `getMatches` but it returns a boolean. Can we rename it?

Reply:
Yes, we can. 
Before and after this PR, `match` returns a enumerator which is used by `BestMatchSpecifier`, while `matches` returns a boolean which is used by `ExactMatchSpecifier`. Those names does not exactly make that distension clear.